qnm_filter/utility.py

```diff
@@ -1,58 +1,47 @@
-#!/usr/bin/env python3
-# -*- coding: utf-8 -*-
+"""Useful functions for calculating and plotting data
 """
-Created on Mon Feb 20 17:55:14 2023
-
-@author: neil
-"""
-
+__all__ = ['parallel_compute', 'plotter']
 
 from joblib import Parallel, delayed
 import matplotlib.pyplot as pl
 import numpy as np
-import time
 
-def parallel_compute(function, x_arr, y_arr, kwargs = {}, show_time = True):
+def parallel_compute(function, M_arr, chi_arr, **kwargs):
     """Parallel computation of a function that takes 2 arguments 
     
     Arguments
     ---------
     function : function
         function that takes 2 args and an arbitrary number of kwargs
-    x_arr : array-like
+    M_arr : array-like
         array of the values of the first arg of the function
-    y_arr : array-like
+    chi_arr : array-like
         array of the values of the second arg of the function
     kwargs : dict
         dictionary of kwargs of the function
-    show_time : bool
-        whether to print time (in minutes) for the calculation
         
     Returns
     ---------
     reshaped_results : ndarray
         2d array of the results with shape (len(x_arr), len(y_arr))
     """
-    tic = time.time()
-    flatten_array = [(i,j) for i in x_arr for j in y_arr]
+    flatten_array = [(i,j) for i in M_arr for j in chi_arr]
     results = Parallel(-1)(delayed(function)(i, j, **kwargs) 
                      for i,j in flatten_array)
-    reshaped_results = np.reshape(results, (len(x_arr), len(y_arr))).T
-    toc = time.time()
-    if show_time:
-        calc_time = (toc-tic)/60
-        print("Calculation time = " + str(np.round(calc_time, 2)) + "mins")
+    reshaped_results = np.reshape(results, (len(M_arr), len(chi_arr))).T
     return reshaped_results
 
-def plotter(X, Y, results, credible_region=None, show = True, save_fig = False, **kwarg_dict):
-    pl.rc('figure', figsize=(6.1, 6.6))
+def plotter(X_grid, Y_grid, results, IMR_result = (np.nan, np.nan), credible_region=None, \
```

Comment:
Move the plotting-related options (`show`, `save_fig `, `fig_size`) to `**kwarg_dict` because we don't always want to specify them.